system-parachains/encointer/src/lib.rs

```diff
@@ -624,16 +570,74 @@ impl pallet_membership::Config<pallet_membership::Instance1> for Runtime {
 	type WeightInfo = weights::pallet_membership::WeightInfo<Runtime>;
 }
 
+/// A `HandleCredit` implementation that naively transfers the fees to the block author.
+/// Will drop and burn the assets in case the transfer fails.
+pub struct AssetsToBlockAuthor<R>(PhantomData<R>);
+
+impl<R> HandleCredit<AccountIdOf<R>, pallet_encointer_balances::Pallet<R>>
+	for AssetsToBlockAuthor<R>
+where
+	R: pallet_authorship::Config + pallet_encointer_balances::Config,
+	AccountIdOf<R>: From<polkadot_primitives::AccountId> + Into<polkadot_primitives::AccountId>,
+{
+	fn handle_credit(credit: Credit<AccountIdOf<R>, pallet_encointer_balances::Pallet<R>>) {
+		if let Some(author) = pallet_authorship::Pallet::<R>::author() {
+			// In case of error: Will drop the result triggering the `OnDrop` of the imbalance.
+			let _ = pallet_encointer_balances::Pallet::<R>::resolve(&author, credit);
```

Comment:
It is not obvious to me how this handles native transfer fees aka KSM.

Reply:
native is handled elsewhere with DealWithFees